bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/internal/WebClientFactoryImpl.java

```diff
@@ -373,7 +350,9 @@ private SslContextFactory createSslContextFactory(@Nullable String endpoint) {
         if (endpoint == null) {
             return createSslContextFactoryFromExtensibleTrustManager();
         } else {
-            return createSslContextFactoryFromTrustManagerProvider(endpoint);
+            SslContextFactory sslContextFactory = new SslContextFactory();
```

Comment:
Even if we are talking about three lines of code. They are duplicate. Should we merge this method with `createSslContextFactoryFromExtensibleTrustManager()` and initialize the `SSLContext` only if `endpoint == null`? 

Or why have we to distinguish here at all.

Reply:
That change makes sense. I'm not that familiar with all this code though.

From what I understand, because of the deprecation, there wouldn't be any instance of a `TrustManagerProvider` anymore. So all code guarded by `trustmanagerProvider != null`, would effectively be dead code.

Maybe @martinvw can also comment on if these deprecated methods are cleaned up the right way?

Even if we don't fully understand how this works... now is the time to clean it up because OH3 will go through some rigorous testing.

Reply:
If the `endpoint` parameter is only used to differentiate between the deprecated  `TrustManagerProvider` and `ExtensibleTrustManager` we might just want to remove the parameter and always use the `ExtensibleTrustManager`?

Reply:
Looks like because of the removal of the other methods, the `endpoint` would now always be `null`.

Reply:
The latest change looks nice. And it should cause no problems because everything was internal.